src/main/java/laika/Laika.java

```diff
@@ -0,0 +1,167 @@
+package laika;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+import java.io.IOException;
+import java.io.FileWriter;
+
+public class Laika {
+
+    public static void modifyTask(ArrayList<Task> taskList, String[] words){
+        int taskNumber = Integer.parseInt(words[1]);
+        if (words[0].equals("mark")) {
+            taskList.get(taskNumber - 1).markAsDone();
+            System.out.println("Laika: Good job! Task has been marked as done.");
+        } else {
+            taskList.get(taskNumber - 1).markAsUndone();
+            System.out.println("Laika: Alright, task has been marked as undone.");
+        }
+        System.out.println(taskList.get(taskNumber-1));
+    }
+
+    public static void deleteTask(ArrayList<Task> taskList,ArrayList<String> lines, String[] words){
+        int taskNumber = Integer.parseInt(words[1]);
+        System.out.println("Laika: Gotcha! I've dealt with this task:" + System.lineSeparator()
+                            + taskList.get(taskNumber-1));
+        taskList.remove(taskNumber-1);
+        lines.remove(taskNumber-1);
+    }
+
+    public static void addTask(ArrayList<Task> taskList, String line) throws LaikaException {
+        if(line.startsWith("todo")){
+            taskList.add(new Todo(line.replaceFirst("todo ","")));
+        }
+        else if (line.startsWith("deadline")) {
+            String[] words = line.split("/");
+            if (words.length != 2){
+                throw new LaikaException();
+            }
+            taskList.add(new Deadline(words[0].replaceFirst("deadline ",""),
+                                           words[1].replaceFirst("by ","")));
+        }
+        else if (line.startsWith("event")) {
+            String[] words = line.split("/");
+            if (words.length != 3){
+                throw new LaikaException();
+            }
+            taskList.add(new Event(words[0].replaceFirst("event ",""),
+                                        words[1].replaceFirst("from ",""),
+                                        words[2].replaceFirst("to ","")));
+        } else {
+            throw new LaikaException();
+        }
```

Comment:
See if you can get rid of the deep nesting here. It will require quite a bit of refactoring